cmd/helm/chart_pull.go

```diff
@@ -32,15 +32,22 @@ This will store the chart in the local registry cache to be used later.
 `
 
 func newChartPullCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
-	return &cobra.Command{
+	var insecureOpt, plainHTTPOpt bool
+	cmd := &cobra.Command{
 		Use:    "pull [ref]",
 		Short:  "pull a chart from remote",
 		Long:   chartPullDesc,
 		Args:   require.MinimumNArgs(1),
 		Hidden: !FeatureGateOCI.IsEnabled(),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ref := args[0]
-			return action.NewChartPull(cfg).Run(out, ref)
+			return action.NewChartPull(cfg).Run(out, ref, insecureOpt, plainHTTPOpt)
```

Comment:
The `Run` function is part of the public API and its signature cannot change. A new function needs to be created. The old one should be listed as deprecated and there should be a note on the new one to clean this up in Helm v4.

This does not apply to internal functions as they are not part of the public API.